.github/workflows/image_build_push.yml

```diff
@@ -0,0 +1,144 @@
+name: docker-image-push-public-dash
+
+on:
+  push:
+    branches: [ main ]
+    
+  workflow_dispatch:
+    inputs:
+      docker_image_tag:
+        description: "Latest Docker image tags passed from e-mission-server repository on image build and push"
+        required: true
+
+env:
+  DOCKER_USER: ${{secrets.DOCKER_USER}}
+  DOCKER_PASSWORD: ${{secrets.DOCKER_PASSWORD}}
+
+jobs:
+  fetch_run_id:
+    runs-on: ubuntu-latest
+
+    outputs:
+      run_id: ${{ steps.get_run_id.outputs.run_id }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with: 
+          python-version: '3.x'
+
+      - name: Install Python dependencies
+        run: |
+         python -m pip install --upgrade pip
+         pip install requests
+
+      - name: Run Python script
+        id: run_script
+        run: |
+          echo "Fetching latest successful run ID from e-misison-server docker image workflow"
+          python .github/fetch_runID.py
+    
+      - name: Get Run ID
+        id: get_run_id
+        run: echo "run_id=${{ steps.run_script.outputs.run_id }}" >> "$GITHUB_OUTPUT"
+
+  fetch_tag:
+      needs: fetch_run_id
+      runs-on: ubuntu-latest
+
+      env:
+        RUN_ID: ${{ needs.fetch_run_id.outputs.run_id }}
+
+      outputs:
+        docker_image_tag: ${{ steps.get_docker_tag.outputs.docker_image_tag }}
+
+      steps:
+      - uses: actions/checkout@v4
+
+      - name: Use Run ID from previous fetch_run_id job
+        run: echo Run ID from previous job ${{ env.RUN_ID }}
+        
+      - name: Download artifact
+        uses: actions/download-artifact@v4
+        with:
+          # TODO: Create a token with basic repo permissions
+          name: docker-image-tag
+          github-token: ${{ secrets.GH_PAT_TAG }}
+          repository: e-mission/e-mission-server
+          run-id: ${{ env.RUN_ID }}
+
+      - name: Print artifact tag
+        id: get_docker_tag
+        run: |
+          cat tag_file.txt
+          docker_image_tag=$(cat tag_file.txt)
+          echo $docker_image_tag
+          echo "docker_image_tag=$(echo $docker_image_tag)" >> $GITHUB_OUTPUT
+  build:
+    needs: fetch_tag
+
+    runs-on: ubuntu-latest
+
+    env:
+      DOCKER_IMAGE_TAG_1: ${{ needs.fetch_tag.outputs.docker_image_tag }}
+      DOCKER_IMAGE_TAG_2: ${{ github.event.inputs.docker_image_tag }}
```

Comment:
nit, future fix: better names

Reply:
Clarification: Better names for the environment variables?